repository/src/main/java/com/walmartlabs/concord/repository/GitCliRepositoryProvider.java

```diff
@@ -51,12 +49,15 @@ public boolean canHandle(String url) {
 
     @Override
     public String getBranchOrDefault(String branch) {
-        return branch != null ? branch : DEFAULT_BRANCH;
+        return branch != null ? branch : GitConstants.DEFAULT_BRANCH;
     }
 
     @Override
-    public String fetch(String uri, String branchOrNull, String commitId, Secret secret, boolean checkRemoteCommitId, Path dst) {
-        String branch = getBranchOrDefault(branchOrNull);
+    public String fetch(String uri, String branchOrNull, String commitId, Secret secret, Path dst) {
+        String branch = branchOrNull;
+        if (commitId == null && branch == null) {
+            branch = GitConstants.DEFAULT_BRANCH;
```

Comment:
Should this really assume DEFAULT_BRANCH is master?  Soon the default branch may be `main` or whatever is name is picked by git / github. Then we'll have 2 defaults (before the change and after).

Perhaps try and find remote HEAD? `git ls-remote --symref https://github.com/walmartlabs/concord.git HEAD`

Reply:
I think we can implement this logic in new PR:) we can try get this info from GH, we can make branch name for repo mandatory, we can put default branch name into configuration...

we use this `DEFAULT_BRANCH` constant not only here...